atlite/data.py

```diff
@@ -124,28 +128,28 @@ def cutout_prepare(cutout, features=slice(None), tmpdir=None, overwrite=False):
         logger.info(f'Calculating and writing with module {module}:')
         missing_features = missing_vars.index.unique('feature')
         ds = get_features(cutout, module, missing_features, tmpdir=tmpdir)
-        # make sure we don't loose any unused dimension by selecting
-        ds = ds[missing_vars.values].assign_coords(ds.coords)
-
-        ds = ds.assign_attrs(**cutout.data.attrs)
         prepared |= set(missing_features)
-        ds = ds.assign_attrs(prepared_features = list(prepared))
-        # convert bool to int for netCDF4 storing
-        ds.attrs.update({k: v if not isinstance(v, bool) else int(v)
-                         for k,v in ds.attrs.items()})
 
+        ds = cutout.data.merge(ds[missing_vars.values])\
+                            .assign_attrs(**cutout.data.attrs, **ds.attrs)
+        ds.attrs = non_bool_dict(cutout.data.attrs)
+        ds.attrs.update({'prepared_features': list(prepared)})
+
+        # write data to tmp file, copy it to original data, this is much saver
+        # than appending variables
+        tstr = next(tmpname())
+        tmp = cutout.path.parent / f'tmp_{tstr}_{cutout.path.stem}.nc'
```

Comment:
Please use `mktemp` instead of the private `_get_candidate_names`, since it also makes sure the candidate does not exist yet:
```suggestion
        # write data to tmp file, copy it to original data, this is much safer
        # than appending variables
        tmp = Path(mktemp(suffix=f"_{cutout.path.stem}.nc, dir=cutout.path.parent))
```

As the next improvement step you should worry about the race condition described in the doc-string of `mktemp`:
> THIS FUNCTION IS UNSAFE AND SHOULD NOT BE USED.  The file name may refer to a file that did not exist at some point, but by the time you get around to creating it, someone else may have beaten you to the punch.

And decide whether you want to use instead the pattern introduced in
https://github.com/PyPSA/atlite/blob/09714d595fc09f2aa1ba79bdd9e60391c582e880/atlite/data.py#L211-L215